prog/dftb+/lib_dftbplus/mainio.F90

```diff
@@ -5294,4 +5301,33 @@ subroutine writeReksDetailedOut1(fd, nGeoSteps, iGeoStep, tMD, tDerivs, &
   end subroutine writeReksDetailedOut1
 
 
+  !> Write cavity information as cosmo file
+  subroutine writeCosmoFile(solvation, species0, speciesNames, coords0, energy)
+
+    !> Instance of the solvation model
+    class(TSolvation), intent(in) :: solvation
+
+    !> Symbols of the species
+    character(len=*), intent(in) :: speciesNames(:)
+
+    !> Species of every atom in the unit cell
+    integer, intent(in) :: species0(:)
+
+    !> Atomic coordinates
+    real(dp), intent(in) :: coords0(:,:)
+
+    !> Total energy
+    real(dp), intent(in) :: energy
+
+    integer :: unit
+
+    select type(solvation)
+    class is (TCosmo)
+      write(stdOut, '(*(a:, 1x))') "Cavity information written to", cosmoFile
```

Comment:
Cool, I am positively surprised too see, that unlimited format repetition finally works with all compilers we use!